[weien02] iP

weien02 opened the pull request:
# 🤖 Step Aside, Duke—Qwerty Has Arrived!  

Introducing **Qwerty**, the to-do list app that doesn’t just track your tasks—it *secretly judges you* for your procrastination.  

## 🌟 What's New?  
- Add tasks to your to-do list  
- Save them to local storage  
- Search tasks with ease  
- **And more**!  

## 🔢 How to Use  
Find this code block in Qwerty.java:  

```java  
public static void main(String[] args) {  
    new Qwerty("./data/Qwerty.txt").run(); // Starts the Qwerty bot with a file path for storage.  
}
```  

Run `main` and voila, Qwerty is ready to go!  

## ✅ To-Do  
- [ ] GUI for normies who don't use CLI 😜  

## 💡 Why Qwerty?  
> "Qwerty is everywhere. Qwerty is watching you."  

## 🔗 Helpful Links  
Check out the full fork [here](https://github.com/weien02/ip)!  

Happy tasking! 🤖

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

weien02 committed:
Level 0. Rename, Greet, Exit

weien02 committed:
Level 1. Echo

weien02 committed:
Level 2. Add, List

weien02 committed:
Mark as Done

weien02 committed:
ToDos, Events, Deadlines

weien02 committed:
Automated Text UI Testing

weien02 committed:
Handle Errors

weien02 committed:
Delete

weien02 committed:
Use Enums

weien02 committed:
Add Level 7. Save

weien02 committed:
Update gitignore

weien02 committed:
Merge branch 'branch-Level-7'

weien02 committed:
Add Level 8 Dates and Times

weien02 committed:
Merge branch 'branch-Level-8'

weien02 committed:
Add A-MoreOOP

weien02 committed:
Add A-Packages

weien02 committed:
Merge branch 'add-gradle-support'

weien02 committed:
Add JUnit Tests

weien02 committed:
Create a JAR File

weien02 committed:
Add A-CodingStandard

weien02 committed:
Merge branch 'branch-A-CodingStandard'

weien02 committed:
Add A-JavaDoc

weien02 committed:
Merge branch 'branch-A-JavaDoc'

weien02 committed:
Add Level 9 Find

weien02 committed:
Merge branch 'branch Level-9'

weien02 committed:
Fix syntax error and style

weien02 committed:
Implement ADT principle for FindCommand

laishuya commented on src/main/java/qwerty/commands/CommandsEnum.java:
```diff
@@ -0,0 +1,34 @@
+    todo, // Command to create a ToDo task.
+    deadline, // Command to create a Deadline task.
+    event, // Command to create an Event task.
+    find; // Command to create a Find task.
```
I like how you used enums for your chatbot as it enforces type safety!

laishuya commented on src/main/java/qwerty/ui/Parser.java:
```diff
@@ -0,0 +1,93 @@
+            case "event":
+                return new EventCommand(desc);
+            case "find":
+                return new FindCommand(desc);
```
Good job for getting the indentation right for switch statements!

laishuya commented on src/main/java/qwerty/task/ToDo.java:
```diff
@@ -0,0 +1,40 @@
+
+import qwerty.exceptions.EmptyTaskNameException;
+
+/**
+ * Represents a ToDo task, which includes a description and completion status.
+ * Inherits from {@link Task} and provides methods for formatting and saving the task.
+ */
+public class ToDo extends Task {
+
+    /**
+     * Constructs a ToDo task with the specified name.
+     *
+     * @param name The name or description of the ToDo task.
+     * @throws EmptyTaskNameException If the task name is empty.
+     */
```
You wrote Javadoc header comments for public classes and constructors, which some people forgot. Good job! 👍

laishuya commented on src/main/java/qwerty/task/Task.java:
```diff
@@ -0,0 +1,62 @@
+    /**
+     * Marks the task as completed.
+     */
+    public void markTaskDone() {
+        this.isDone = true; // Sets task status to completed.
+    }
+
+    /**
+     * Unmarks the task as completed.
+     */
+    public void unmarkTaskDone() {
+        this.isDone = false; // Sets task status to not completed.
+    }
```
Could the // comments be unnecessary here?

laishuya commented on src/main/java/qwerty/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,39 @@
+ */
+public class DeadlineCommand extends BotCommand {
+
+    private String desc; // Description of the Deadline task.
```
Do you think desc could be made "final" as well? I'm wondering if I should do that for my iP.

panomia01 commented on src/main/java/qwerty/task/Deadline.java:
```diff
@@ -0,0 +1,55 @@
+
+    private static final DateTimeFormatter INPUT_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+    private static final DateTimeFormatter OUTPUT_FORMATTER = DateTimeFormatter.ofPattern("MMM dd yyyy, h:mma");
+    private LocalDateTime by; // Deadline date and time.
```
perhaps u can consider changing to a noun

panomia01 commented on src/main/java/qwerty/task/Task.java:
```diff
@@ -0,0 +1,62 @@
+     *
+     * @return A string representing the Task in a save-friendly format.
+     */
+    public String toSaveString() {
```
its nitpicking but can change to saveToString() instead such that it is in a sensible order

panomia01 commented on src/main/java/qwerty/ui/Storage.java:
```diff
@@ -0,0 +1,100 @@
+     *
+     * @return An ArrayList of tasks read from the file.
+     */
+    public ArrayList<Task> readListFromFile() {
```
readListFromFile -> readTaskListFromFile

nit: just making it easier to understand

panomia01 commented on src/main/java/qwerty/task/Event.java:
```diff
@@ -0,0 +1,47 @@
+public class Event extends Task {
+
+    private String from; // Start time of the event.
+    private String to; // End time of the event.
```
Can consider changing these variable names to better nouns like timeStart, timeEnd for easy understanding

panomia01 reviewed (commented):
LGTM overall, only have some small naming conventions to change. Structure was easy to read and code is tidy

weien02 committed:
Set up JavaFX

weien02 committed:
Complete Level 10

weien02 committed:
Merge branch 'branch-Level-10'

weien02 committed:
Remove redundant css files

weien02 committed:
Add A-Assertions

Use assertion to check if user input is empty.

weien02 committed:
Add A-CodingQuality

Modify method/class header comments to align with the coding standard, especially in the phrasing of the overview statement.

Ensure method/class header comments follow the specified format in the coding standard.

Following the standard improves readability, consistency, and maintainability of the codebase.

weien02 committed:
Merge pull request #2 from weien02/branch-A-Assertions

Add A-Assertions

weien02 committed:
Merge pull request #3 from weien02/branch-A-CodeQuality

Add A-CodingQuality

weien02 committed:
Add D-Loans

Add support managing loan records

Implement functionality for managing loan records, including items lent or owed to colleagues and friends.

weien02 committed:
Pull from remote repo

weien02 committed:
Fix coding standard

Code did not fully comply with checkstyle rules, particularly regarding constant naming conventions.

Update code to use SCREAMING_SNAKE_CASE for final variables.

Ensuring adherence to checkstyle improves code readability, maintainability, and consistency across the codebase.

weien02 committed:
Refactor code

Some methods were too long and had weak SLAP.

Refactor the methods by extracting helper methods to separate logic.

Applying SLAP improves code readability, maintainability, and makes future modifications easier.

weien02 committed:
Update README

The README contained outdated references to Duke instead of Qwerty, which could cause confusion.

Update the README to reflect Qwerty instead of Duke.

Ensuring the correct project name improves clarity and maintains consistency with the current implementation.

weien02 committed:
Add JUnit test

The createCommand method in Parser did not have a test case to verify behavior for unknown commands.

Add a JUnit test case to check if createCommand throws UnknownCommandException when an unknown command is input.

Including this test ensures that the method handles invalid input correctly and maintains the expected behavior.

weien02 committed:
Add title

The title bar of the application did not reflect the correct project name.

Add "Qwerty" to the title bar of the application.

This ensures consistency with the project name and provides a clearer user interface.

weien02 committed:
Add User Guide

The project lacked a user guide for proper usage instructions.

Add a user guide to provide clear instructions for users on how to interact with the application.

This ensures users can easily understand and navigate the application.

weien02 committed:
Update User Guide

The user guide contained incorrect instructions for running the JAR file.

Update the user guide with correct instructions on how to run the JAR file.

This ensures users can correctly execute the application from the JAR file.

weien02 committed:
Update User Guide

The user guide contained stray text that could cause confusion for users.

Update the user guide to remove any stray text and ensure clarity.

This improves the overall quality and readability of the user guide.

weien02 committed:
Improve UX

The bot lacked personality, and the UI could be improved for a better user experience. The user guide also needed updates to reflect these changes.

Give the bot a personality, improve the UI, and update the user guide accordingly.

These changes make interactions more engaging, the interface more user-friendly, and the documentation more accurate.

weien02 committed:
Update bot responses

Some bot responses did not fit the correct context or sounded unnatural.

Update bot phrases using suggestions from ChatGPT to ensure they match the intended context and sound more natural.

This improves user experience by making interactions smoother and more engaging.
